tools/run_tests/python_utils/bazel_report_helper.py

```diff
@@ -0,0 +1,252 @@
+
+    os.makedirs(report_path, exist_ok=True)
+
+    bazel_wrapper_filename = os.path.join(report_path, 'bazel_wrapper')
+    bazel_wrapper_bat_filename = bazel_wrapper_filename + '.bat'
+    bazel_rc_filename = os.path.join(report_path, 'bazel_rc')
+
+    # put xml reports in a separate directory if requested by GRPC_TEST_REPORT_BASE_DIR
+    report_base_dir = os.getenv('GRPC_TEST_REPORT_BASE_DIR', None)
+    xml_report_path = os.path.abspath(
+        os.path.join(report_base_dir, report_path
+                    ) if report_base_dir else report_path)
+    os.makedirs(xml_report_path, exist_ok=True)
+
+    failing_report_filename = os.path.join(xml_report_path, 'sponge_log.xml')
+    success_report_filename = os.path.join(xml_report_path,
+                                           'success_log_to_rename.xml')
+
+    if _platform_string() == 'windows':
+        workspace_status_command = 'tools/remote_build/workspace_status_kokoro.bat'
+    else:
+        workspace_status_command = 'tools/remote_build/workspace_status_kokoro.sh'
+
+    # generate RC file with the bazel flags we want to use apply.
+    # Using an RC file solves problems with flag ordering in the wrapper.
+    # (e.g. some flags need to come after the build/test command)
+    with open(bazel_rc_filename, 'w') as f:
+        f.write('build --invocation_id="%s"\n' % invocation_id)
+        f.write('build --workspace_status_command="%s"\n' %
+                workspace_status_command)
+
+    # generate "failing" and "success" report
+    # the "failing" is named as "sponge_log.xml", which is the name picked up by sponge/resultstore
+    # so the failing report will be used by default (unless we later replace the report with
+    # one that says "success"). That way if something goes wrong before bazel is run,
+    # there will at least be a "failing" target that indicates that (we really don't want silent failures).
+    with open(failing_report_filename, 'w') as f:
+        f.write(
+            _generate_junit_report_string(report_suite_name,
+                                          invocation_id,
+                                          success=False))
+    with open(success_report_filename, 'w') as f:
+        f.write(
+            _generate_junit_report_string(report_suite_name,
+                                          invocation_id,
+                                          success=True))
+
+    # generate the bazel wrapper for linux/macos
+    with open(bazel_wrapper_filename, 'w') as f:
+        intro_lines = [
+            '#!/bin/bash',
+            'set -ex',
+            '',
+            'tools/bazel --bazelrc="%s" "$@" || FAILED=true' %
+            bazel_rc_filename,
+            '',
+        ]
+
+        if upload_results:
+            upload_results_lines = [
+                '# Sleep to let ResultStore finish writing results before querying',
+                'sleep 60',
```

Comment:
We might want to make this configurable. E.g., a script-wide constant at the top. Also... if the script is going to be disposable, it probably isn't that useful to add comments in it?

Reply:
done.